todo_app/app.py

```diff
@@ -1,54 +1,120 @@
+        
+        client.parse_request_body_response(token_response.text)
+
+        user_response = requests.get('https://api.github.com/user', headers={
+            'Accept': 'application/json', 
+            'Authorization': f"token {token_response.json()['access_token']}"
+            })
+        
+        login_user(User(user_response.json()['login']))        
+        return redirect(url_for('index'))
+    
+    def get_user_role():             
+        if (current_app.config.get('LOGIN_DISABLED') or current_user.user_id == admin_user_id):                        
+            return ROLE
+        else:            
+            return 'readonly'
+    
     @app.route('/', methods=['Get'])
+    @login_required
     def index():
-        tasks = trello.get_all_tasks()        
-        task_view_model = view_model.ViewModel(tasks)
+        tasks = taskDb.get_all_tasks()
+        user_role = get_user_role()   
+        task_view_model = view_model.ViewModel(tasks,user_role)
         return render_template('index.html', view_model=task_view_model)
 
+
     @app.route('/add_todo', methods=['Post'])
+    @login_required
     def add_todo_task():
-        trello.create_todo_task(request.form.get('title'))
+        user_role = get_user_role()
+        if (user_role == ROLE):
+            taskDb.create_todo_task(request.form.get('title'))
+            return redirect('/')
+        else:
+            return abort(403)
+        
         return redirect('/')
 
     @app.route('/move_to_doing/<todo_task_id>', methods=['Post'])
+    @login_required
     def move_to_doing(todo_task_id):
-        trello.move_to_doing(todo_task_id)
-        return redirect('/')
+        user_role = get_user_role()
+        if (user_role == ROLE):
+            taskDb.move_to_doing(todo_task_id)
+            return redirect('/')
+        else:
+            return abort(403)
 
     @app.route('/move_to_done/<task_id>', methods=['Post'])
+    @login_required
     def move_to_done(task_id):
-        trello.move_to_done(task_id)
-        return redirect('/')
+        user_role = get_user_role()
+        if (user_role == ROLE):
+            taskDb.move_to_done(task_id)
+            return redirect('/')
+        else:
+            return abort(403)
 
     @app.route('/delete_task/<task_id>', methods=['Post'])
-    def delete_task(task_id):
-        trello.delete_task(task_id)
-        return redirect('/')
-
-    #if __name__ == '__main__':
-    #    app.run()
-    
-    #return app
-    if __name__ == '__main__':
-        #app.debug = True
-        app.run(host='0.0.0.0', port=5000)  
+    @login_required
+    def remove_task(task_id):
+        user_role = get_user_role()
+        if (user_role == ROLE):
+            taskDb.delete_task(task_id)
+            return redirect('/') 
+        else:
+            return abort(403)
 
     return app
 
+app = create_app(taskDb)
+app.secret_key = secret_key
+login_manager.init_app(app)
```

Comment:
Why is this done here? When running the app locally, this means create_app is called twice as Flask uses that by default, and that means the login_manager has not been initialised for apps created directly through that function. Instead we should do this initialisation inside the create_app function